src/Pages/Applications/DevtronApps/service.types.ts

```diff
@@ -55,3 +56,77 @@ export interface ResourceConfig {
 export interface EnvConfigDTO {
     resourceConfig: ResourceConfig[]
 }
+
+export enum AppEnvDeploymentConfigType {
+    PUBLISHED_ONLY = 'PublishedOnly',
+    DRAFT_ONLY = 'DraftOnly',
+    PUBLISHED_WITH_DRAFT = 'PublishedWithDraft',
+    PREVIOUS_DEPLOYMENTS = 'PreviousDeployments',
+    DEFAULT_VERSION = 'DefaultVersion',
+}
+
+export interface DraftMetadata {
+    appId: number
+    envId: number
+    resource: number
```

Comment:
```suggestion
    resourceId: number
```

Reply:
api response key

Reply:
I think we should ask them to update this. Please confirm the same with @vivek-devtron